docs/development/local_setup.md

```diff
@@ -0,0 +1,82 @@
+# Preparing the Local Development Setup (Mac OS X)
+
+Conceptionally, the Node Controller Manager is designated to run in a container within a Pod inside a Kubernetes cluster. However, if you want to develop it, you may want to work locally with the Node Controller Manager without building a Docker image and deploying it to a cluster each and every time. That means that the Node Controller Manager runs outside a Kubernetes cluster which requires providing a [Kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/) in your local filesystem and point the Node Controller Manager to it when running it (see below). 
+
+Although the following installation instructions are for Mac OS X, similar alternate commands could be found for any other Linux distributions.
+
+## Installing Golang environment
+Install the latest version of Golang (at least `v1.8.3` is required) by using [Homebrew](https://brew.sh/):
+
+```bash
+$ brew install golang
+```
+
+Make sure to set your `$GOPATH` environment variable properly (conventionally, it points to `$HOME/go`).
+
+For your convenience, you can add the `bin` directory of the `$GOPATH` to your `$PATH`: `PATH=$PATH:$GOPATH/bin`, but it is not necessarily required.
+
+In order to perform linting on the Go source code, please install [Golint](https://github.com/golang/lint):
+
+```bash
+$ go get -u github.com/golang/lint/golint
+```
+
+In order to perform tests on the Go source code, please install [Ginkgo](https://onsi.github.io/ginkgo/) and [Gomega](http://onsi.github.io/gomega/). Please make yourself familiar with both frameworks and read their introductions after installation:
+
+```bash
+$ go get -u github.com/onsi/ginkgo/ginkgo
+$ go get -u github.com/onsi/gomega
+```
+
+We use [Dep](https://github.com/golang/dep) for managing Golang package dependencies. Please install it:
+
+```bash
+$ brew install dep
+```
+
+## Installing `git`
+We use `git` as VCS which you need to install:
+
+```bash
+$ brew install git
+```
+
+## Installing Docker (Optional)
+In case you want to build Docker images for the Node Controller Manager you have to install Docker itself. We recommend using [Docker for Mac OS X](https://docs.docker.com/docker-for-mac/) which can be downloaded from [here](https://download.docker.com/mac/stable/Docker.dmg).
+
+## Installing `gcloud` SDK (Optional)
+In case you have to create a new release or a new hotfix of the Node Controller Manager you have to push the resulting Docker image into a Docker registry. Currently, we are using the Google Container Registry (this could change in the future). Please follow the offical installation instructions [from Google](https://cloud.google.com/sdk/downloads#mac).
```

Comment:
it seems you are switching to use DockerHub in this commit so it seems you are not using Google Container Registry

Reply:
I have removed gcloud reference and added a section on creating an account on docker hub as optional